src/Model/Contact.php

```diff
@@ -1459,11 +1459,15 @@ public static function getIdForURL($url, $uid = 0, $no_update = false, $default
 			$contact_id = $contact["id"];
 
 			// Update the contact every 7 days
-			$update_contact = ($contact['updated'] < DateTimeFormat::utc('now -7 days'));
```

Comment:
Cheeky suggestion:
```php
$update_contact = in_array($contact['network'], Protocol::NATIVE_SUPPORT)
    && ($contact['updated'] < DateTimeFormat::utc('now -7 days')
        || empty($contact['avatar']));
```

Reply:
There is a second statement some lines below that is changing the variable as well. So I enclosed them both by the "if" statement.

Reply:
I can't see anything else modifying `$update_contact`, I think my short version is correct, although maybe harder to read.

Reply:
Ah, now I saw that you already enclosed the "empty" check as well.

Well ... this is **really** harder to read.